src/app/shared/components/header/header.html

```diff
@@ -0,0 +1,68 @@
+<mat-toolbar
+  aria-label="Main navigation"
+  color="primary"
+  role="banner">
+  <div class="header-container">
+    <a
+      aria-label="FoxFlow Home"
+      class="logo"
+      routerLink="/">
+      <img
+        alt="FoxFlow Logo"
+        height="48"
+        ngSrc="assets/images/logo.png"
+        width="48" />
+    </a>
+    <span class="spacer"></span>
+    @if (!isAuthenticated && (isMobile$ | async) === false) {
+    <div class="nav-links">
+      <a
+        aria-label="Log in"
+        mat-button
+        routerLink="/login"
+        routerLinkActive="active">
+        {{ 'AUTH.LOGIN' | translate }}
+      </a>
+      <a
+        aria-label="Register"
+        mat-button
+        routerLink="/register"
+        routerLinkActive="active">
+        {{ 'AUTH.REGISTER' | translate }}
+      </a>
+    </div>
+    } @if (!isAuthenticated && (isMobile$ | async)) {
+    <button
+      aria-label="Open navigation menu"
+      mat-icon-button
+      type="button"
+      [matMenuTriggerFor]="menu">
+      <mat-icon>menu</mat-icon>
+    </button>
+    }
+    <mat-menu #menu="matMenu">
+      <a
+        aria-label="Log in"
+        mat-menu-item
+        routerLink="/login">
+        {{ 'AUTH.LOGIN' | translate }}
+      </a>
+      <a
+        aria-label="Register"
+        mat-menu-item
+        routerLink="/register">
+        {{ 'AUTH.REGISTER' | translate }}
+      </a>
+    </mat-menu>
+    <app-language-switcher />
+    <mat-slide-toggle
+      aria-label="Toggle dark/light theme"
+      class="icon-only-toggle"
+      matTooltip="Toggle dark/light theme"
+      matTooltipPosition="above"
+      [checked]="isDarkMode"
+      (change)="toggleTheme()">
+      <mat-icon>{{ isDarkMode ? 'brightness_4' : 'brightness_7' }}</mat-icon>
```

Comment:
Please use the corresponding icons for light and dark mode ui elements

```suggestion
      <mat-icon>{{ isDarkMode ? 'dark_mode' : 'light_mode' }}</mat-icon>
```

Reply:
Thank you for pointing this out! Fixes are applied.